components/experimental/src/units/converter.rs

```diff
@@ -89,14 +88,13 @@ impl OffsetConverter {
 /// Also, it cannot convert between two units that are not single, such as `meter` to `foot-and-inch`.
 #[derive(Debug)]
 pub(crate) struct ProportionalConverter {
-    // TODO(#4554): Implement a New Struct `IcuRatio` to Encapsulate `Ratio<BigInt>`.
     /// The conversion rate between the input and output units.
-    pub(crate) conversion_rate: Ratio<BigInt>,
+    pub(crate) conversion_rate: IcuRatio,
 }
 
 impl ProportionalConverter {
     /// Converts the given value from the input unit to the output unit.
-    pub fn convert(&self, value: &Ratio<BigInt>) -> Ratio<BigInt> {
-        value * &self.conversion_rate
+    pub fn convert(&self, value: &IcuRatio) -> IcuRatio {
+        value.clone() * self.conversion_rate.clone()
```

Comment:
question: where does all this cloning come from?

Reply:
the IcuRatio could not implement `Copy` trait

Reply:
Please avoid calling `.clone()` for short-lived values. It doesn't make sense here: you are allocating two ratios and then immediately throwing them away to make a third.

Reply:
Done, but I am not happy to call lthe functions on value not references

https://github.com/unicode-org/icu4x/pull/4799/commits/e8755e57d56e4615f014b0d6593a1eea0284f27f

Reply:
I reverted, we could not remove clone because of the following error.

```
cannot move out of `self.conversion_rate` which is behind a shared reference
move occurs because `self.conversion_rate` has type `IcuRatio`, which does not implement the `Copy` traitrustc[Click for full compiler diagnostic](rust-analyzer-diagnostics-view:/diagnostic%20message%20%5B0%5D?0#file%3A%2F%2F%2FUsers%2Fyouniesmahmoud%2Fi18n%2Ficu4x%2Fcomponents%2Fexperimental%2Fsrc%2Funits%2Fconverter.rs)
IcuRatio
```

and we could not implement Copy for IcuRatio for the folloiwng error

```
the trait `std::marker::Copy` cannot be implemented for this typerustc[Click for full compiler diagnostic](rust-analyzer-diagnostics-view:/diagnostic%20message%20%5B0%5D?0#file%3A%2F%2F%2FUsers%2Fyouniesmahmoud%2Fi18n%2Ficu4x%2Fcomponents%2Fexperimental%2Fsrc%2Funits%2Fratio.rs)
ratio.rs(22, 21): this field does not implement `std::marker::Copy`
ratio.rs(22, 21): the `std::marker::Copy` impl for `Ratio<BigInt>` requires that `BigInt: std::marker::Copy`
core::marker
macro Copy
```

Reply:
That's the default error when Rust thinks it needs an owned type, but you should be able to perform your arithmetic using borrowed types. You might be missing some impls. @robertbastian can help you debug this.

Reply:
Okay